tcp-native/src/main/java/org/ballerinalang/stdlib/tcp/SecureSocketUtils.java

```diff
@@ -0,0 +1,183 @@
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.crypto.Cipher;
+import javax.crypto.EncryptedPrivateKeyInfo;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static java.util.regex.Pattern.CASE_INSENSITIVE;
+import static javax.crypto.Cipher.DECRYPT_MODE;
+
+/**
+ * SecureSocket utility functions.
+ */
+public class SecureSocketUtils {
+
+    public static KeyStore truststore(String certificateLocation) throws CertificateException, IOException,
+            KeyStoreException, NoSuchAlgorithmException {
+        // load the certificate
+        InputStream is = new FileInputStream(certificateLocation);
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+        X509Certificate cert = (X509Certificate) cf.generateCertificate(is);
+
+        // store the certificate key truststore
+        KeyStore ts = KeyStore.getInstance(KeyStore.getDefaultType());
+        ts.load(null); // create truststore on the fly
+        ts.setCertificateEntry("client", cert); // store the certificate
+
+        return ts;
+    }
+
+    public static KeyStore keystore(String certificateChainFileLocation, String privateKeyLocation,
+                                    Optional<String> keyPassword) throws IOException, GeneralSecurityException {
+        PKCS8EncodedKeySpec encodedKeySpec =
+                SecureSocketUtils.readPrivateKey(new File(privateKeyLocation), Optional.empty());
+        PrivateKey key = null;
+        boolean rasInvalidKeySpecException = false;
+
+        try {
+            KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+            key = keyFactory.generatePrivate(encodedKeySpec);
+        } catch (InvalidKeySpecException ignore) {
+            rasInvalidKeySpecException = true;
+        }
+
+        // if RSA failed then try with DSA
+        if (rasInvalidKeySpecException) {
+            KeyFactory keyFactory = KeyFactory.getInstance("DSA");
+            key = keyFactory.generatePrivate(encodedKeySpec);
+        }
+
+
+        List<X509Certificate> certificateChain =
+                SecureSocketUtils.readCertificateChain(new File(certificateChainFileLocation));
+        if (certificateChain.isEmpty()) {
+            throw new CertificateException("Certificate file does not contain any certificates: "
+                    + certificateChainFileLocation);
+        }
+
+        KeyStore keyStore = KeyStore.getInstance("JKS");
+        keyStore.load(null, "secret".toCharArray());
```

Comment:
Can't we use null as the `Keystore` password? If not for the moment shall we extract the password as a constant?

Reply:
+1

Reply:
keystore password changed to null, keyEntry password replaced with constant